R/print.R

```diff
@@ -6,55 +6,59 @@
 #' @param parent_ref The calling function.
-#' @param expression The deparsed expression (if present) on which `ic()` was called.
-#' @param value The result of evaluating `expression`.
+#' @param deparsed_expression The deparsed expression (if present) on which `ic()` was called. If
+#'   missing (default value), only evaluation context is printed.
+#' @param value The result of evaluating `deparsed_expression`. If expression is missing (default
+#' value), this argument should also be missing.
 #'
 #' @return The function is called for its side effect (printing to the console) but it also returns
 #'   its output string, invisibly.
 #'
 #' @keywords internal
-ic_print <- function(loc, parent_ref, expression, value) {
-
-  context_string <- loc
+#' @importFrom rlang expr_deparse is_missing missing_arg
+#' @importFrom utils capture.output str
+#' @importFrom glue glue
+#' @importFrom cli cli_alert_info
+ic_print <- function(loc, parent_ref, deparsed_expression = missing_arg(), value = missing_arg()) {
+  # TODO: both loc and deparsed_expression (if present, ofc) are
 
   # TODO: I'm not certain at this stage that we will never get a zero-char `loc` passed in. There is
   #       probably a better way of handling this, but for now this will do.
-  if (nchar(context_string) == 0) {
-    context_string <- "<unknown>"
-  }
-  expression_string <- NULL
+  context_string <- if (nchar(loc) == 0) "<unknown>" else loc
 
   # Next, are we printing a calling function?
   if (!is.null(parent_ref)) {
-    parent_ref <- deparse(parent_ref)
-    context_string <- glue::glue("{{.fn {parent_ref}}} in {context_string}")
+    parent_ref <- expr_deparse(parent_ref)
+    context_string <- glue("{{.fn {parent_ref}}} in {context_string}")
   }
 
+  expression_string <- NULL
+
   # Formatting result
-  if (!is.null(expression) && !is.null(value)) {
+  if (!is_missing(deparsed_expression)) {
     # We want to print a one-line summary for complex objects like lists and data frames.
     #
     # TODO: Taking the first line of output from `str()` is a quick way of getting this but it
     #       doesn't produce great output (try passing in a `lm()` object - ugly). It would be nice
     #       to fix this at some point.
-    str_res <- trimws(utils::capture.output(utils::str(value)))[[1L]]
-    expression_string <- glue::glue("{{.var {expression}}}: {str_res}")
+    str_res <- trimws(capture.output(str(value)))[[1]]
+    expression_string <- glue("{{.var {deparsed_expression}}}: {str_res}")
   }
 
   # We need to check what options are set to decide what to print - whether to include the context
   # or not.
-  if (!is.null(expression_string)) {
-    if (getOption("icecream.include.context")) {
-      output <- glue::glue("{context_string} | {expression_string}")
-    } else {
-      output <- expression_string
-    }
-  } else {
-    output <- context_string
-  }
-
+  #
+  # TODO: Shall icecream.include.context be reanamed to something like
```

Comment:
Good point - I went with `icecream.include.context` to match the Python version which uses `includeContext` (see [the "configuration" section of the original's README](https://github.com/gruns/icecream#configuration). TBH I don't have a strong opinion either way. Do you think there's any particular value in remaining consistent with the original? In either case I think the README could be updated to more explicitly say that empty `ic()` calls always print the context.

Reply:
I don't think it's necessary to stick so closely to the original code -- we're inevitably working in a different language and will differ in some aspects, so I don't see any objection to improving some more language-independent aspects as well. I'm a big proponent of verbosity, even if it involves longer names and so one. Users will have to look up the exact name of the option anyway.

Reply:
Yep agree. Opened #2 - will fix it once this has been merged